File: API/gis/src/main/java/pt/isel/ps/gis/hypermedia/siren/components/subentities/Entity.java
package pt.isel.ps.gis.hypermedia.siren.components.subentities;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.HashMap;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({"class", "rel", "properties", "href"})
public class Entity {
    @JsonProperty(value = "class")
    public String[] klass;
    public String[] rel;
    public HashMap<String, Object> properties;
    public String href;

    public Entity(String[] klass, String[] rel, HashMap<String, Object> properties, String href) {
        this.klass = klass;
        this.rel = rel;
        this.properties = properties;
        this.href = href;
    }

    public String[] getKlass() {
        return klass;
    }

    public String[] getRel() {
        return rel;
    }

    public String getHref() {
        return href;
    }
}

File: CD/projeto/AppsAndroid/gis/app/src/main/java/ps/leic/isel/pt/gis/hypermedia/siren/subentities/Entity.kt
package ps.leic.isel.pt.gis.hypermedia.siren.subentities

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

@JsonIgnoreProperties(ignoreUnknown = true)
class Entity(@field:JsonProperty(value = "class") var klass: Array<String>?, var rel: Array<String>,
             var properties: HashMap<String, Any>?, var actions: Array<Action>?, var links: Array<Link>?)
